wled00/e131.cpp

```diff
@@ -225,11 +225,17 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
           if (e131_data[dataOffset+3]   != seg.intensity) seg.intensity = e131_data[dataOffset+3];
           if (e131_data[dataOffset+4]   != seg.palette)   seg.setPalette(e131_data[dataOffset+4]);
 
-          uint8_t segOption = (uint8_t)floor(e131_data[dataOffset+5]/64.0);
-          if (segOption == 0 && (seg.mirror  || seg.reverse )) {seg.setOption(SEG_OPTION_MIRROR, false); seg.setOption(SEG_OPTION_REVERSED, false);}
-          if (segOption == 1 && (seg.mirror  || !seg.reverse)) {seg.setOption(SEG_OPTION_MIRROR, false); seg.setOption(SEG_OPTION_REVERSED,  true);}
-          if (segOption == 2 && (!seg.mirror || seg.reverse )) {seg.setOption(SEG_OPTION_MIRROR,  true); seg.setOption(SEG_OPTION_REVERSED, false);}
-          if (segOption == 3 && (!seg.mirror || !seg.reverse)) {seg.setOption(SEG_OPTION_MIRROR,  true); seg.setOption(SEG_OPTION_REVERSED,  true);}
+          uint8_t segOption = (uint8_t)floor(e131_data[dataOffset+5]);
```

Comment:
I do not understand why use `floor()` here. There's no float/double involved.

Reply:
In current version: It devides 255 dmx channel values into 4 parts `e131_data[dataOffset+5]/64.0` by `value/64.0` it gets 0..4 float, so floor part asures that `segOption` is always 0|1|2|3, simple to choose which of the 4 options to select by sending 0..255 values (as described in the docs). To add more than 4 options within one dmx channel (=255 values), you should use `value/(255/X)` where X is number of options per channel. For example if you want 8 options use `value/32`. In DMX language this is called macros. Segment option channel is a DMX macro channel devided into 4 macros. Of cause `uint8_t segOption = (uint8_t)floor(e131_data[dataOffset+5]);` does not make any sence because it is not float! This could be implemented differently of cause, but for readability I decided to go with uint value -> devide -> floor to choose one macro out of 255 possible values.

Reply:
Why not just use integer division? In the end it is the same.

Reply:
Oops, I forgot to remove that call to `float` when I switched from division to bitshifting when handling segment options. This has been changed in `f848f57`.

Reply:
Sorry, I'm not familiar with bit shift syntax. @demophoon maybe use integer division as @blazoncek suggested instead, just for better code readability.  But I don't mind if it does the same .. I think with bit shift, you don't need that extra variable `segOption`, couldn't you just use `e131_data[dataOffset+5]` directly?

Reply:
Or just use `e131_data[dataOffset+5] & 0b00001000` or similar for particular check.

Reply:
I've made both of these suggestions in [97e20fe](https://github.com/Aircoookie/WLED/pull/3616/commits/97e20fe3571371ae27e9a47441df75c0fc3c07aa).